front/src/applications/operationalStudies/hooks/useScenarioData.ts

```diff
@@ -122,17 +169,42 @@ const useScenarioData = (scenario: ScenarioResponse, infra: InfraWithState) => {
   // TODO Paced train : Adapt this to accept paced trains in issue https://github.com/OpenRailAssociation/osrd/issues/10613
   useAutoUpdateProjection(infra, trainsIds, trainScheduleSummaries);
 
+  // TODO Paced trains : remove this useEffect in https://github.com/OpenRailAssociation/osrd/issues/10791
   useEffect(() => {
-    const formattedRawTrainSchedules: TrainScheduleResultWithTrainId[] = trainSchedulesResults.map(
-      (trainSchedule) => ({
-        ...trainSchedule,
-        id: formatEditoastTrainIdToTrainScheduleId(trainSchedule.id),
-      })
-    );
-    const sortedTrainSchedules = sortBy(formattedRawTrainSchedules, 'start_time');
-    setTrainSchedules(sortedTrainSchedules);
-  }, [trainSchedulesResults]);
+    if (!showPacedTrains) {
+      const formattedRawTrainSchedules: TrainScheduleResultWithTrainId[] =
+        trainSchedulesResults.map((trainSchedule) => ({
+          ...trainSchedule,
+          id: formatEditoastTrainIdToTrainScheduleId(trainSchedule.id),
+        }));
+      const sortedTrainSchedules = sortBy(formattedRawTrainSchedules, 'start_time');
+      setTrainSchedules(sortedTrainSchedules);
+    }
+  }, [trainSchedulesResults, showPacedTrains]);
+
+  useEffect(() => {
+    // TODO Paced trains : remove the if and extra depth in https://github.com/OpenRailAssociation/osrd/issues/10791
+    if (showPacedTrains) {
+      const formattedRawTrainSchedules: TrainScheduleResultWithTrainId[] =
+        trainSchedulesResults.map((trainSchedule) => ({
+          ...trainSchedule,
+          id: formatEditoastTrainIdToTrainScheduleId(trainSchedule.id),
+        }));
+      const formattedRawPacedTrains: PacedTrainResultWithPacedTrainId[] = pacedTrainsResults.map(
+        (trainSchedule) => ({
+          ...trainSchedule,
+          id: formatEditoastTrainIdToPacedTrainId(trainSchedule.id),
+        })
+      );
+      const sortedTimatableItems = sortBy(
+        [...formattedRawTrainSchedules, ...formattedRawPacedTrains],
+        'start_time'
+      );
+      setTimetableItems(sortedTimatableItems);
+    }
```

Comment:
If you have this above:
```
  const trainSchedules: TrainScheduleResultWithTrainId[] = useMemo(
    () =>
      (fetchedTrainSchedulesResults || []).map((trainSchedule) => ({
        ...trainSchedule,
        id: formatEditoastTrainIdToTrainScheduleId(trainSchedule.id),
      })),
    [fetchedTrainSchedulesResults]
  );
```
then, you can simply do (pacedTrains will be empty if !showPacedTrains): 
```suggestion
      const sortedTimetableItems = sortBy([...trainSchedules, ...pacedTrains], 'start_time');
      setTimetableItems(sortedTimetableItems);
    }
```

Then, you can remove the previous useEffect

Reply:
Done